header.php

```diff
@@ -4,18 +4,43 @@
 
     check_cors();
 
-    $cmd = "echo $((`cat /sys/class/thermal/thermal_zone0/temp | cut -c1-2`))";
-    $output = shell_exec($cmd);
-    $celsius = str_replace(array("\r\n","\r","\n"),"", $output);
-    $fahrenheit = round(str_replace(["\r\n","\r","\n"],"", $output*9./5)+32);
+    // Try to get temperature value from different places (OS dependent)
+    if(file_exists("/sys/class/thermal/thermal_zone0/temp"))
+    {
+        $output = rtrim(file_get_contents("/sys/class/thermal/thermal_zone0/temp"));
+    }
+    elseif (file_exists("/sys/class/hwmon/hwmon0/temp1_input"))
+    {
+        $output = rtrim(file_get_contents("/sys/class/hwmon/hwmon0/temp1_input"));
+    }
+    else
+    {
+        $output = "";
+    }
 
-    if(isset($setupVars['TEMPERATUREUNIT']))
+    // Test if we succeeded in getting the temperature
+    if(is_numeric($output))
     {
-        $temperatureunit = $setupVars['TEMPERATUREUNIT'];
+        $celsius = intVal($output)*1e-3;
+        $fahrenheit = ($celsius*9./5)+32.0;
+
+        if(isset($setupVars['TEMPERATUREUNIT']))
+        {
+            $temperatureunit = $setupVars['TEMPERATUREUNIT'];
+        }
+        else
+        {
+            $temperatureunit = "C";
+        }
+        // Override temperature unit setting if it is changed via Settings page
+        if(isset($_POST["tempunit"]))
+        {
+            $temperatureunit = $_POST["tempunit"];
+        }
     }
     else
     {
-        $temperatureunit = "C";
+        $celsius = -273.16;
```

Comment:
Why is $celsius set to -273.16? Perhaps set it to null and check if it's null later.
http://php.net/manual/en/language.types.null.php